### PR TITLE
chore: default cache client testing

### PR DESCRIPTION
**Pull request by pgautier404**
This commit adds testing for `CacheClient` instances created using the `NewCacheClientWithDefaultCache` constructor. It focuses on happy path test cases to ensure that the default cache is used when `CacheName` is empty or omitted and that the default cache is overridden when `CacheName` is specified.

### CODE REVIEW

File: momento/set_test.go [hunk 1]
```diff
@@ -190,24 +200,56 @@ var _ = Describe("Set methods", func() {
 			},
 			Entry(
 				"when elements are strings",
+				DefaultClient,
+				[]Value{String("hello"), String("world"), String("!"), String("␆")},
+				[]string{"hello", "world", "!", "␆"},
+				[][]byte{[]byte("hello"), []byte("world"), []byte("!"), []byte("␆")},
+			),
+			Entry(
+				"when elements are bytes",
```
Comment (eaddingtonwhite) [inline momento/set_test.go:209]: is it ok to have multiple entries in table test with same name? For example this one `"when elements are bytes"` is repeated on line 237. Should we rename 2nd set of tests here so Entries are named like `"when elements are bytes with default cache"`?

File: momento/set_test.go [hunk 2]
```diff
@@ -278,23 +327,27 @@ var _ = Describe("Set methods", func() {
...
 			},
-			Entry("as string", String("#5"), 9),
-			Entry("as bytes", Bytes([]byte("#5")), 9),
-			Entry("unmatched", String("notvalid"), 10),
+			Entry("as string", DefaultClient, String("#5"), 9),
+			Entry("as bytes", DefaultClient, Bytes([]byte("#5")), 9),
+			Entry("unmatched", DefaultClient, String("notvalid"), 10),
+			Entry("as string", WithDefaultCache, String("#5"), 9),
+			Entry("as bytes", WithDefaultCache, Bytes([]byte("#5")), 9),
+			Entry("unmatched", WithDefaultCache, String("notvalid"), 10),
```
Comment (eaddingtonwhite) [inline momento/set_test.go:335]: same comment here on duplicate entry names in table test.

File: momento/set_test.go [hunk 3]
```diff
@@ -306,9 +359,12 @@ var _ = Describe("Set methods", func() {
...
 			},
-			Entry("as strings", getElements(5), 5),
-			Entry("as bytes", []Value{Bytes("#3"), Bytes("#4")}, 8),
-			Entry("unmatched", []Value{String("notvalid")}, 10),
+			Entry("as strings", DefaultClient, getElements(5), 5),
+			Entry("as bytes", DefaultClient, []Value{Bytes("#3"), Bytes("#4")}, 8),
+			Entry("unmatched", DefaultClient, []Value{String("notvalid")}, 10),
+			Entry("as strings", WithDefaultCache, getElements(5), 5),
+			Entry("as bytes", WithDefaultCache, []Value{Bytes("#3"), Bytes("#4")}, 8),
+			Entry("unmatched", WithDefaultCache, []Value{String("notvalid")}, 10),
```
Comment (eaddingtonwhite) [inline momento/set_test.go:367]: duplicate entry names

File: momento/set_test.go [hunk 4]
```diff
@@ -366,9 +429,12 @@ var _ = Describe("Set methods", func() {
...
 			},
-			Entry("all hits", []Value{String("#1"), String("#2"), String("#3")}, []bool{true, true, true}),
-			Entry("all misses", []Value{String("not"), String("this"), String("time")}, []bool{false, false, false}),
-			Entry("a mixture", []Value{String("not"), String("#2"), String("time")}, []bool{false, true, false}),
+			Entry("all hits", DefaultClient, []Value{String("#1"), String("#2"), String("#3")}, []bool{true, true, true}),
+			Entry("all misses", DefaultClient, []Value{String("not"), String("this"), String("time")}, []bool{false, false, false}),
+			Entry("a mixture", DefaultClient, []Value{String("not"), String("#2"), String("time")}, []bool{false, true, false}),
+			Entry("all hits", WithDefaultCache, []Value{String("#1"), String("#2"), String("#3")}, []bool{true, true, true}),
+			Entry("all misses", WithDefaultCache, []Value{String("not"), String("this"), String("time")}, []bool{false, false, false}),
+			Entry("a mixture", WithDefaultCache, []Value{String("not"), String("#2"), String("time")}, []bool{false, true, false}),
```
Comment (eaddingtonwhite) [inline momento/set_test.go:437]: duplicate entry names

File: momento/sorted_set_test.go [hunk 5]
```diff
@@ -119,14 +128,19 @@ var _ = Describe("SortedSet", func() {
...
+		Entry("Empty cache name", DefaultClient, "", sharedContext.CollectionName, InvalidArgumentError),
+		Entry("Blank cache name", DefaultClient, "  ", sharedContext.CollectionName, InvalidArgumentError),
+		Entry("Empty collection name", DefaultClient, sharedContext.CacheName, "", InvalidArgumentError),
+		Entry("Blank collection name", DefaultClient, sharedContext.CacheName, "  ", InvalidArgumentError),
+		Entry("Non-existent cache", DefaultClient, uuid.NewString(), uuid.NewString(), NotFoundError),
+		Entry("Empty cache name", WithDefaultCache, "", sharedContext.CollectionName, InvalidArgumentError),
+		Entry("Blank cache name", WithDefaultCache, "  ", sharedContext.CollectionName, InvalidArgumentError),
+		Entry("Empty collection name", WithDefaultCache, sharedContext.CacheName, "", InvalidArgumentError),
+		Entry("Blank collection name", WithDefaultCache, sharedContext.CacheName, "  ", InvalidArgumentError),
+		Entry("Non-existent cache", WithDefaultCache, uuid.NewString(), uuid.NewString(), NotFoundError),
```
Comment (eaddingtonwhite) [inline momento/sorted_set_test.go:140]: duplicate entry names

File: momento/sorted_set_test.go [hunk 6]
```diff
@@ -966,8 +1011,10 @@ var _ = Describe("SortedSet", func() {
 					Expect(fetchResp.ValueBytesElements()).To(Equal(expected))
 				}
 			},
-			Entry("string value", String("aString"), 42.0, []SortedSetBytesElement{{Value: []byte("aString"), Score: 42}}),
-			Entry("bytes value", Bytes("aString"), 42.0, []SortedSetBytesElement{{Value: []byte("aString"), Score: 42}}),
+			Entry("string value", DefaultClient, String("aString"), 42.0, []SortedSetBytesElement{{Value: []byte("aString"), Score: 42}}),
+			Entry("bytes value", DefaultClient, Bytes("aString"), 42.0, []SortedSetBytesElement{{Value: []byte("aString"), Score: 42}}),
+			Entry("string value", WithDefaultCache, String("aString"), 42.0, []SortedSetBytesElement{{Value: []byte("aString"), Score: 42}}),
+			Entry("bytes value", WithDefaultCache, Bytes("aString"), 42.0, []SortedSetBytesElement{{Value: []byte("aString"), Score: 42}}),
```
Comment (eaddingtonwhite) [inline momento/sorted_set_test.go:1017]: duplicate entry names

**Review by eaddingtonwhite — COMMENTED** [review]
@pgautier404 looking good. I think maybe just in set_test.go copypasta'd some of the table test entry need to give uniquenames to each case? Other then that looking good. Thanks for doing this!

File: momento/set_test.go (hunk 1 above)
Comment (pgautier404) [reply momento/set_test.go:209]: Good catch. It doesn't really hurt anything but would make them a little harder to track down if they failed. Will update.

**Review by eaddingtonwhite — APPROVED** [review]

**Pull request merged**